actions/service.go

```diff
@@ -380,12 +380,8 @@ func (s *Service) GetTaskResult(ctx context.Context, repositoryID string, runID
 	return res.(*TaskResult), nil
 }
 
-func (s *Service) ListRunResults(ctx context.Context, repositoryID string, branchID *string, after string) (RunResultIterator, error) {
-	return NewDBRunResultIterator(ctx, s.DB, defaultFetchSize, repositoryID, branchID, nil, after), nil
-}
-
-func (s *Service) ListCommitRunResults(ctx context.Context, repositoryID string, commitID string) (RunResultIterator, error) {
-	return NewDBRunResultIterator(ctx, s.DB, defaultFetchSize, repositoryID, nil, &commitID, ""), nil
+func (s *Service) ListRunResults(ctx context.Context, repositoryID string, branchID, commitID *string, after string) (RunResultIterator, error) {
```

Comment:
General note: we can pass optionalBranchID and optionalCommitID as string and not *string because an empty string is not a valid branch or commit.

Reply:
Done.